File: WorkoutTrackerAPI/Data/Models/UserModels/ExerciseRecordGroup.cs
using WorkoutTrackerAPI.Data.Models.WorkoutModels;

namespace WorkoutTrackerAPI.Data.Models.UserModels;

public class ExerciseRecordGroup : IDbModel
{
    public long Id { get; set; }

    public long ExerciseId { get; set; }
    public Exercise? Exercise { get; set; }

    public long WorkoutRecordId { get; set; }
    public WorkoutRecord? WorkoutRecord { get; set; }

    public IEnumerable<ExerciseRecord> ExerciseRecords { get; set; } = null!;
}

File: WorkoutTracker/src/app/shared/models/exercise-record-group.ts
import { DbModel } from "src/app/shared/models/db-model";
import { ExerciseType } from "src/app/exercises/models/exercise-type";
import { ModelWeight } from "./model-weight";
import { ExerciseRecord } from "src/app/exercise-records/exercise-record";

export interface ExerciseRecordGroup extends DbModel {
    sets: number;
    weight: ModelWeight;

    exerciseId: number;
    exerciseName: string;
    exerciseType: ExerciseType;

    exerciseRecords: ExerciseRecord[];
}